src/main/java/com/linecorp/armeria/client/RemoteInvokerFactory.java

```diff
@@ -127,4 +127,49 @@ public void close() {
             eventLoopGroup.shutdownGracefully().syncUninterruptibly();
         }
     }
+
+    private static final class ConfiguredDnsNameResolverGroup extends DnsNameResolverGroup {
+
+        private static final DnsServerAddresses DEFAULT_DNS_SERVER_ADDRS;
+        private static final int DEFAULT_MAX_TRIES;
+
+        static {
+            // FIXME(trustin): Workaround for https://github.com/netty/netty/pull/4541
+            //                 Simplify when fixed.
+            List<InetSocketAddress> defaultList = DnsServerAddresses.defaultAddressList();
+            List<InetSocketAddress> doubledList = new ArrayList<>(defaultList.size() * 2);
+            doubledList.addAll(defaultList);
+            doubledList.addAll(defaultList);
+            final int start = defaultList.size();
```

Comment:
Took me many minutes to finally realize your intent, how about this (performance doesn't matter much since it's static initializers)?

```java
List<InetSocketAddress> defaultList = new ArrayList<>(DnsServerAddresses.defaultAddressList());
List<InetSocketAddress> doubledList = new ArrayList<>(defaultList.size() * 2);
doubledList.addAll(defaultList);
// Add list of servers, rotated left by 1, so netty does not always cycle through the same servers
Collections.rotate(defaultList, 1);
doubledList.addAll(defaultList);
```